src/App.tsx

```diff
@@ -1,12 +1,64 @@
 import React from 'react';
 import './App.scss';
+import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRed } from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+export interface State {
+  goods: Good[],
+
+}
+
+export class App extends React.Component {
+  state = {
+    goods: [],
+  };
+
+  render() {
+    return (
+      <div className="container">
+        Dynamic list of goods
+        <div>
+          <button
+            className="button"
+            type="button"
+            onClick={async () => {
+              const goods = await getAll();
+
+              this.setState({ goods });
+            }}
+          >
+            show all
+          </button>
```

Comment:
You can also move handler to separate method which takes callback as a parameter

Reply:
Agree with @DionStrzh